virtme/guest/virtme-sshd-script

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+#
+# Initialize ssh server for remote connections (option `--server ssh`)
+
+if [ ! -n "${virtme_ssh_user}" ]; then
+    echo "ssh: virtme_ssh_user is not defined" >&2
+    exit 1
+fi
+
+# Generate authorized_keys adding the host's user public keys, but only if
+# the rootfs is mounted in read-only.
+SSH_HOME=$(getent passwd "${virtme_ssh_user}" | cut -d: -f6)
+if [ ! -e "${SSH_HOME}" ]; then
+    # Setup an arbitrary ssh location, just to be able to start sshd.
+    SSH_HOME=/run/ssh
+fi
+if grep ' / ' /proc/mounts | grep -q ' ro,'; then
```

Comment:
It's a detail, but it feels strange to modify files only if the rootfs is in read-only :-)

Would it not make more sense to grep on `/home overlay rw,`?

Reply:
Haha yes, it's quite obscure if you're not familiar with the filesystem layout. :smile:  I'll change the logic to check for the rw overlay.

Reply:
Actually, I take it back... trying to figure out if the cache dir is in an overlay isn't trivial, because potentially it can be outside of /home. Basically what I care about is preventing any changes to the host fs, and that happens only when `--rw` is passed, and the most effective way from the guest to check if `--rw` was specified is to check if the rootfs has been mounted in `rw` or `ro`.

Maybe I can keep this check and add more details to explain why we're doing this check.

Reply:
Thank you for the update. I think the current version is good, no need to do other modifications.

Just in case, if all you want to check is if overlayfs is used, you can use:

```shell
OVERLAYFS="794c7630"
if [ "$(stat -f -c "%t" "${SSH_HOME}")" = "${OVERLAYFS}" ]; then
```

(you could use `%T` which should print `overlayfs`, but it doesn't work with busybox apparently -- you can use [`OVERLAYFS_SUPER_MAGIC`](https://elixir.bootlin.com/linux/v6.12.6/source/include/uapi/linux/magic.h#L38) instead)

Reply:
Looks like a good change for a separate PR. :)

Reply:
done :-)

https://github.com/arighi/virtme-ng/pull/211